dev/inc/CppWinRTHelpers.h

```diff
@@ -217,6 +217,31 @@ inline PropertyChanged_revoker RegisterPropertyChanged(winrt::DependencyObject c
 }
 
 
+inline bool SetFocus(winrt::DependencyObject const& object, winrt::FocusState focusState)
```

Comment:
>SetFocus [](start = 12, length = 8)

If running on a new enough OS, you can use FocusManager.TryFocusAsync which takes a DO so you don't have to special-case every element type.

What's here works well today, just throwing that out there in case you want to future-proof the code a bit more.

Reply:
Thanks, good suggestion. Except TryFocusAsync doesn't tell me synchronously if the element could be focused. Hmm...

Reply:
It does the vast majority of the time - we return synchronously whenever possible, it only ends up being truly async if the call must be async like a cross-process focus call into Out-Of-Process WebView.

---
In reply to: [270983114](https://github.com/Microsoft/microsoft-ui-xaml/pull/511#discussion_r270983114) [](ancestors = 270983114)

Reply:
So you could kick off the potentially-async operation.  If it's immediately synchronously completed, return the result it gave you - if not, return true since it's likely to succeed.

An example where it can async fail is if you try to focus an out-of-process WebView that doesn't contain anything focusable inside, when WebView will then return focus to XAML.  That's pretty rare.

---
In reply to: [270984620](https://github.com/Microsoft/microsoft-ui-xaml/pull/511#discussion_r270984620) [](ancestors = 270984620,270983114)

Reply:
That's what I ended up with too. Glad you agree with the approach. :)